qiita_db/artifact.py

```diff
@@ -1171,6 +1162,19 @@ def path_cleaner(fps):
             qdb.sql_connection.TRN.add(sql, sql_args, many=True)
             qdb.sql_connection.TRN.execute()
 
+        # to avoid deleting potentially necessary files, we are going to add
+        # that check after the previous transaction is commited
+        if to_delete_fps:
+            for x in self.filepaths:
+                if x['fp'] in to_delete_fps:
+                    to_delete_fps.remove(x['fp'])
+
+            for fp in to_delete_fps:
+                if isfile(fp):
+                    remove(fp)
+                else:
+                    rmtree(fp)
```

Comment:
rmtree() can take a callback function to handle errors or ignore failures to delete. Just to confirm, is it better to just have it raise an Error here instead of using a handler or ignoring it?

Reply:
it has been fine so far ... but if you have specific examples of what to add and why, please let me know.

Reply:
True. I just realized path_cleaner() is using the same parameters. I was thinking it might be good to avoid an Error in that part of the code, but it hasn't caused issues so far, and I'd need to look deeper into it in order to provide a specific example.